### PR TITLE
Start final method statics

### PR DESCRIPTION
**Pull request by azdavis**
This begins the impl of static checks for final methods.

### Motivation
To report errors statically.

### Test plan
See included automated tests.

There are some things marked TODO in the tests. This is pending further future work.

### CODE REVIEW

**Comment by jez** [comment]
cc @DarkDimius you might want to take a look at this too.

We're planning on landing the final static checks in stages to break up the implementation (rather than doing it all statically in one shot).

File: definition_validator/validator.cc [hunk 1]
```diff
@@ -278,9 +279,9 @@ class ValidateWalk {
 public:
     unique_ptr<ast::ClassDef> preTransformClassDef(core::Context ctx, unique_ptr<ast::ClassDef> classDef) {
         auto sym = classDef->symbol;
-        validateAbstract(ctx.state, sym);
-        validateTStructNotGrandparent(ctx.state, sym);
         auto singleton = sym.data(ctx)->lookupSingletonClass(ctx);
+        validateTStructNotGrandparent(ctx.state, sym);
```
Comment (DarkDimius) [inline definition_validator/validator.cc]: ooc: why did the order change?

File: test/testdata/resolver/final_method.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 # typed: true
+# disable-fast-path: true
```
Comment (DarkDimius) [inline test/testdata/resolver/final_method.rb]: ooc: why?

File: definition_validator/validator.cc (hunk 1 above)
Comment (azdavis) [reply definition_validator/validator.cc]: Just to line up the calls to validateAbstract one after the other so it looks nicer.

**Pull request merged**